src/ray/core_worker/transport/direct_actor_transport.cc

```diff
@@ -109,17 +111,20 @@ Status CoreWorkerDirectActorTaskSubmitter::SubmitTask(TaskSpecification task_spe
   } else {
     // Do not hold the lock while calling into task_finisher_.
     task_finisher_.MarkTaskCanceled(task_id);
-    std::shared_ptr<rpc::RayException> creation_task_exception = nullptr;
+    rpc::ErrorType error_type;
+    std::shared_ptr<rpc::RayException> creation_task_exception;
```

Comment:
Why unique ptr here? I think we allow this ptr to be copied in the following codes, like PendingTaskFailed(currently won't be copied, but I think we should allow this in case of changing interface in the future.).

Reply:
Shared pointer means shared ownership, and it doesn’t really make sense in this function since the function will lose ownership right away. We have a code guideline to make ownership more clear, so I think unique pointer or optional is probably a better choice here. Shared pointer also has unnecessary atomic operation under the hood when returned.